2.evaluate_models/classify_genotypes/nbconverted/genotype_classification_evaluation.py

```diff
@@ -0,0 +1,188 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Calculate evaluation Plate and Datasplit evaluation results
+# Evaluation results include confusion matrices, pr curves, precision, recall, f1-score, and accuracy.
+# This occurs for each plate across all splits.
+
+# In[1]:
+
+
+import pathlib
+from collections import defaultdict
+
+import pandas as pd
+from joblib import load
+from sklearn.metrics import (
+    accuracy_score,
+    confusion_matrix,
+    f1_score,
+    precision_recall_curve,
+    precision_score,
+    recall_score,
+)
+
+# ## Find the root of the git repo on the host system
+
+# In[2]:
+
+
+# Get the current working directory
+cwd = pathlib.Path.cwd()
+
+if (cwd / ".git").is_dir():
+    root_dir = cwd
+
+else:
+    root_dir = None
+    for parent in cwd.parents:
+        if (parent / ".git").is_dir():
+            root_dir = parent
+            break
+
+# Check if a Git root directory was found
+if root_dir is None:
+    raise FileNotFoundError("No Git root directory found.")
+
+
+# ## Define paths
+
+# ### Input
+
+# In[3]:
+
+
+data_path = pathlib.Path(f"{root_dir}/1.train_models/classify_genotypes/data")
+
+evaldf = pd.read_parquet(f"{data_path}/nf1_model_pre_evaluation_results.parquet")
+le = load(f"{data_path}/trained_nf1_model_label_encoder.joblib")
+
+
+# ### Outputs
+
+# In[4]:
+
+
+eval_path = pathlib.Path("model_evaluation_data")
+eval_path.mkdir(parents=True, exist_ok=True)
+
+
+# In[5]:
+
+
+gene_column = "true_genotype"
+
+def down_sample_by_genotype(_df):
+    """
+    Parameters
+    ----------
+    _df: Pandas Dataframe
+        The data to be downsampled by the gene_column column.
+
+    Returns
+    -------
+        The data down-sampled by genotype.
+    """
+
+    min_gene = _df[gene_column].value_counts().min()
+    return (_df.groupby(gene_column, group_keys=False)
+            .apply(lambda x: x.sample(n=min_gene, random_state=0))
+            )
+
+
+# ## Calculate evaluation metrics
+
+# In[6]:
+
+
+# Define evaluation metrics
+eval_mets = {
+    met: defaultdict(list) for met in
+    ("metrics", "precision_recall", "confusion_matrix")
```

Comment:
I recommend being more explicit on what "metrics" means. This current name a bit vague and it would make sense to give a more accurate name since the other two files tell you exactly what data is in it.

Reply:
Ah, it's intentionally vague to capture multiple metrics. Is there another name you would recommend?

Reply:
Something simple like `accuracy_f1_scores` is just fine unless there are other metrics.

Reply:
It does contain other metrics, I may just add a comment mentioning it